src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder().append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int sum = 0;
+
+            for (String datum : data) {
+                String date = datum.split(" ")[0];
+                String worker = datum.split(" ")[1];
+                int hours = Integer.parseInt(datum.split(" ")[2]);
+                int rate = Integer.parseInt(datum.split(" ")[3]);
```

Comment:
can we save the result of `split` in one variable not to do it all over again? it takes time

Reply:
fixed